gst/tensor_decoder/tensordec.c

```diff
@@ -1004,7 +1013,7 @@ gst_tensordec_set_caps (GstBaseTransform * trans,
   silent_debug_caps (incaps, "from incaps");
   silent_debug_caps (outcaps, "from outcaps");
 
-  if (g_strcmp0 (self->mode, "image_labeling") == 0) {
+  if (self->mode == IMAGE_LABELING) {
```

Comment:
line 1016 ~ 1029 is unnecessary.

line 1031 ~ 1040, this lines are to configure tensor info actually. 
IMO, it is better to set in-place mode false in line 1042 with the condition.
if mode==labeling then in-place=false
else if mode==video then in-place=!self->add_padding
else in-place=true

Reply:
I refactored codes as your recommand